### DISCUSSION TITLE
Searching in the Clock Tree Viewer?

### DISCUSSION THREAD

**Discussion by oharboe** [The-OpenROAD-Project/OpenROAD#3755] [General]
Is there a way to search for a net or a port in the Clock Tree Viewer in the GUI?

I can zoom in then hover on one of the leafs and I can see what it connects to, very time-consuming though...

![image](https://github.com/The-OpenROAD-Project/OpenROAD/assets/2798822/dfbdfa85-61df-4671-a6d5-c81714e3de16)

**Comment by maliberty** [comment]
Not to my knowledge (@gadfort )

**Reply by oharboe** [reply]
Silly question: is there a text version of the clock tree report? It would be searchable :-)

**Reply by gadfort** [reply]
@maliberty no, but it could be a nice improvement

**Reply by oharboe** [reply]
Even a text base dump would be nice...

**Reply by maliberty** [reply]
There is report_clock_skew

**Reply by oharboe** [reply]
Doesn't give out all the information in the Clock Tree Viewer...

**Reply by maliberty** [reply]
nope but its what we have today

**Reply by oharboe** [reply]
Is all the information available via the Python or Tcl interface? Perhaps just a utility fn in Tcl/Python?

**Reply by maliberty** [reply]
Ultimately it all comes from sta.  I'm not sure exactly which pieces of data you need but likely you can do it in tcl (probably not in Python yet).
